purge-docker-cluster.yml

```diff
@@ -1,43 +1,447 @@
 ---
 # This playbook purges a containerized Ceph cluster
-# It removes: packages, configuration files and ALL THE DATA
+# It removes: packages, containers, configuration files and ALL THE DATA
 
-- hosts:
-  - mons
-  - osds
+- name: confirm whether user really meant to purge the cluster
+
+  hosts:
+    - localhost
+
+  vars_prompt:
+    - name: ireallymeanit
+      prompt: Are you sure you want to purge the cluster?
+      default: 'no'
+      private: no
+
+  tasks:
+  - name: exit playbook, if user did not mean to purge cluster
+    fail:
+      msg: >
+        "Exiting purge-docker-cluster playbook, cluster was NOT purged.
+         To purge the cluster, either say 'yes' on the prompt or
+         or use `-e ireallymeanit=yes` on the command line when
+         invoking the playbook"
+    when: ireallymeanit != 'yes'
+
+
+- name: retrieve variables
+
+  hosts:
+    - all
```

Comment:
Good point. I assumed the user would have only these ceph cluster hosts in their inventory file, but I know that's not the case. However, how do we know what the variables *_group_name are without including some variable files first? In other words, how do we restrict the hosts without knowing what the hosts are yet? Unless you are thinking we define the *_group_name variables in this file the way the purge-cluster.yml playbook does it. But that seems to be duplicating variables and restricting the user's flexibility.